src/core/crypto_tc.c

```diff
@@ -390,12 +391,19 @@ int32_t Crypto_TC_Do_Encrypt_PLAINTEXT(uint8_t sa_service_type, SecurityAssociat
 
         /* Get Key */
         ekp = key_if->get_key(sa_ptr->ekid);
-        if (ekp == NULL)
+        akp = key_if->get_key(sa_ptr->akid);
+        if (ekp == NULL || akp == NULL)
```

Comment:
Believe this needs to be an AND?

Reply:
Line 395 was meant to check both keys for null returns of get_key. Just thought I would knock them both out in the same line.

Reply:
On 395 isn't it valid if one of them isn't NULL? Therefore the other could be null

Reply:
Yes. A key should only be NULL if the ekid or akid are less than 0 or greater than NUM_KEYS, therefore making it invalid. In any case, the ekid AND akid should be within this range. If either are NULL, then it should throw an error. Correct me if I'm wrong though.